File: csharp/phone-number/PhoneNumber.cs
using System;
using System.Text.RegularExpressions;

public class PhoneNumber
{
    public static string Clean(string phoneNumber)
    {
        if (!new Regex(@"^ *\+?1? *([ \(-\.]*[2-9]\d\d[ \)-\.]*){2}\d{4} *$").IsMatch(phoneNumber))
            throw new ArgumentException("Invalid phone number!", nameof(phoneNumber));

        return new Regex(@"(?!\d).").Replace(phoneNumber, "")[^10..];
    }
}

File: solutions/julia/phone-number/2/phone-number.jl
function clean(phone_number)
    !occursin(r"^ *\+?1? *([ \(-\.]*[2-9]\d\d[ \)-\.]*){2}\d{4} *$", phone_number) && throw(ArgumentError(""))
    replace(phone_number, r"(?!\d)." => "")[(end-9):end]
end